### PR TITLE
S2688 FN: Do not raise for IsExpression since it works as expected

### LINKED ISSUE #6289: S2688 FN: Do not raise for IsExpression since it works as expected
### Description

FN for S2688 (CompareNaN).

Using C#11 list patterns, the rule is not raising an issue.

### Repro steps

```csharp
        object[] numbers = { 1, 2, double.NaN };

        if (numbers is [1, 2, double.NaN]) // FN
        {
            Console.WriteLine("Test1");
        }
```

### Expected behavior

The rule should raise an issue, since `numbers is [..,double.NaN]` is implicitly comparing a value with `double.NaN`.

### Actual behavior

The rule does not raise an issue.

### PR DESCRIPTION
**Pull request by gregory-paidis-sonarsource**
Fixes #6289

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/CompareNaN.cs [hunk 1]
```diff
@@ -33,27 +33,21 @@ namespace SonarAnalyzer.Rules.CSharp
...
         private const string DiagnosticId = "S2688";
-        private const string MessageFormat = "Use {0}.IsNaN() instead.";
+        private const string MessageFormat = "{0}";
+        private const string MessageFormatEquality = "Use {0}.IsNaN() instead.";
+        private const string MessageFormatComparison = "Do not compare a number with {0}.NaN.";
 
         private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
         private static readonly Dictionary<KnownType, string> KnownTypeAliasMap = new()
         {
-            { KnownType.System_Byte, "byte" },
```
Comment (gregory-paidis-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/CompareNaN.cs:41]: Removed some types from here, that do not have a `.NaN` property.

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]

**Comment by csaba-sagi-sonarsource** [comment]
Something is wrong with the Syntax.xml file. It has 9663 lines changed. Please take a look at what is the problem.

**Comment by gregory-paidis-sonarsource** [comment]
> Something is wrong with the Syntax.xml file. It has 9663 lines changed. Please take a look at what is the problem.

Indeed, i reseted the changes locally and pushed with force the same commit, should be okay, please take a look again.

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
I left a few comments, after addressing that it should be good to go!

**Comment by csaba-sagi-sonarsource** [comment]
Please update the PR title to reflect that it is not going to fix the list pattern case it was not a FN.

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/Syntax.xml [hunk 2]
```diff
@@ -1997,7 +1997,18 @@
       <Kind Name="UnderscoreToken"/>
     </Field>
   </Node>
-  <Node Name="DeclarationPatternSyntax" Base="PatternSyntax">
+  <Node Name="ListPatternSyntax" Base="PatternSyntax">
```
[the remark this answers is not in the document]
Comment (gregory-paidis-sonarsource) [reply analyzers/src/SonarAnalyzer.CFG/ShimLayer/Syntax.xml]: you are correct, this remained from a previous implementation by mistake, same with the expression bodied method comment.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**